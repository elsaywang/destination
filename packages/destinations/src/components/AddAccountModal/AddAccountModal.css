:local(.sections) {
    margin: 10px auto 10px auto;
}

:local(.dialog) {
    width: 60%;
    max-width: 600px;
}

:local(.iframe) {
    background-color: white;
    border-radius: 3px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    height: 300px;
}

:local(.iframe_default_text) {
    text-align: center;
    position: relative;
    top: 50%;
    transform: translateY(-50%);
    font-size: 2em;
    font-weight: 200;
}

:local(.email_input) {
    width: 100%;
}

:local(.platform_dropdown_section) {
    display: flex;
    align-items: center;
}

:local(.platform_dropdown_section) > * {
    margin: 5px;
}

p {
    margin: 3px;
}

:local(.menuOption){
  min-width: 120px;
}
