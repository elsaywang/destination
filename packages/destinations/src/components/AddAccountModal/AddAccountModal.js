import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import ModalTrigger from '@react/react-spectrum/ModalTrigger';
import Dialog from '@react/react-spectrum/Dialog';
import Button from '@react/react-spectrum/Button';
import Dropdown from '@react/react-spectrum/Dropdown';
import { Menu, MenuItem } from '@react/react-spectrum/Menu';
import Textfield from '@react/react-spectrum/Textfield';
import requiredIf from 'react-required-if';
import styles from './AddAccountModal.css';
import { getPlatformOptions } from '../../constants/integratedPlatformsOptions';

/*
    Currently Spectrum doesn't propogate prop changes to ModalTrigger children because they are
    doing a bunch of props mutating stuff. As far as I can tell this isn't a bug, their spec is
    that the modal works for static content only. So we must define a stateful component within
    the ModalTrigger in order to get state changes to render to screen.
*/

class ModalContentHackForSpectrum extends Component {
    state = { selectedPlatform: null, confirmDisabled: true };

    handleContactEmailsChange = e => {
        //TODO: add contacts email event
    };

    render() {
        const dropDownOptionMessage =
            this.state.selectedPlatform ||
            (this.props.contactOnlyMode ? this.props.platform : 'Select a Platform');

        const emailContactsPlaceholder = this.props.contactEmails.length
            ? this.props.contactEmails.slice(0).join(`,   `)
            : 'Enter one or more email addresses separate with commas';

        return (
            <Dialog
                role="dialog"
                className={styles.dialog}
                title={this.props.title}
                confirmDisabled={!this.props.contactOnlyMode}
                confirmLabel={this.props.contactOnlyMode ? 'Add' : ' N/A'}
                onConfirm={() => this.props.contactOnlyMode}
                cancelLabel="Cancel"
                {...this.props}>
                <div className={classnames(styles.platform_dropdown_section, styles.sections)}>
                    <span>People-Based Platform *</span>
                    <Dropdown
                        onSelect={selectedPlatform =>
                            this.setState({ selectedPlatform, confirmDisabled: false })
                        }>
                        <Button
                            disabled={this.props.contactOnlyMode}
                            variant="action"
                            className={styles.menuOption}>
                            {dropDownOptionMessage}
                        </Button>
                        <Menu>
                            {getPlatformOptions().map(platform => (
                                <MenuItem
                                    className={styles.menuOption}
                                    key={platform}
                                    value={platform}>
                                    {platform}
                                </MenuItem>
                            ))}
                        </Menu>
                    </Dropdown>
                </div>

                {!this.props.contactOnlyMode && (
                    <div className={styles.sections}>
                        <p>Authorization</p>
                        <div className={styles.iframe}>
                            <p className={styles.iframe_default_text}>
                                {this.state.selectedPlatform
                                    ? `${this.state.selectedPlatform} goes here`
                                    : 'Select a Platform'}
                            </p>
                        </div>
                    </div>
                )}

                {this.props.contactOnlyMode && (
                    <div className={styles.sections}>
                        <p>Contact Emails For Expiring Notification</p>
                        <Textfield
                            placeholder={emailContactsPlaceholder}
                            quiet
                            className={styles.email_input}
                            defaultValue={this.props.contactEmails}
                            onChange={this.handleContactEmailsChange}
                        />
                    </div>
                )}
            </Dialog>
        );
    }
}

const AddAccountModal = ({
    contactOnlyMode,
    message,
    quiet,
    className,
    title,
    triggerIcon,
    platform,
    contactEmails,
}) => (
    <ModalTrigger>
        <Button
            label={message}
            variant="primary"
            modaltrigger
            quiet={quiet}
            className={className}
            icon={triggerIcon}
            data-test='add-account-modal-button'
        />
        <ModalContentHackForSpectrum
            role="dialog"
            contactOnlyMode={contactOnlyMode}
            title={title}
            platform={platform}
            contactEmails={contactEmails}
        />
    </ModalTrigger>
);

AddAccountModal.defaultProps = {
    quiet: false,
    title: 'Add Account',
    icon: null,
    platform: '',
    contactEmails: [],
};

AddAccountModal.propTypes = {
    contactOnlyMode: PropTypes.bool,
    title: PropTypes.string,
    message: PropTypes.string,
    variant: PropTypes.string,
    className: PropTypes.string,
    triggerIcon: PropTypes.node,
    platform: requiredIf(PropTypes.string, ({ contactOnlyMode }) => contactOnlyMode),
    contactEmails: requiredIf(PropTypes.array, ({ contactOnlyMode }) => contactOnlyMode),
};

export default AddAccountModal;
