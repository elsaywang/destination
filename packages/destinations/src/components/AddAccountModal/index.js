import AddAccountModal from './AddAccountModal';

export default AddAccountModal;
