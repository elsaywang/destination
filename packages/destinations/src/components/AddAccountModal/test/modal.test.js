import React from 'react';
import { shallow } from 'enzyme';
import AddAccountModal from '../AddAccountModal';

describe('<AddAccountModal/> component', () => {
    describe('rendering', () => {
        const wrapper = shallow(<AddAccountModal />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
