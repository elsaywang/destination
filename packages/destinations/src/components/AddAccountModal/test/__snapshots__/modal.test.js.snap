// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddAccountModal/> component rendering matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddAccountModal
    contactEmails={Array []}
    icon={null}
    platform=""
    quiet={false}
    title="Add Account"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Button
          autoFocus={false}
          block={false}
          className={undefined}
          data-test="add-account-modal-button"
          disabled={false}
          element="button"
          icon={undefined}
          invalid={false}
          label=""
          logic={false}
          modaltrigger={true}
          quiet={false}
          selected={false}
          variant="primary"
/>,
        <ModalContentHackForSpectrum
          contactEmails={Array []}
          contactOnlyMode={undefined}
          platform=""
          role="dialog"
          title="Add Account"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "block": false,
          "className": undefined,
          "data-test": "add-account-modal-button",
          "disabled": false,
          "element": "button",
          "icon": undefined,
          "invalid": false,
          "label": "",
          "logic": false,
          "modaltrigger": true,
          "quiet": false,
          "selected": false,
          "variant": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "contactEmails": Array [],
          "contactOnlyMode": undefined,
          "platform": "",
          "role": "dialog",
          "title": "Add Account",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Button
            autoFocus={false}
            block={false}
            className={undefined}
            data-test="add-account-modal-button"
            disabled={false}
            element="button"
            icon={undefined}
            invalid={false}
            label=""
            logic={false}
            modaltrigger={true}
            quiet={false}
            selected={false}
            variant="primary"
/>,
          <ModalContentHackForSpectrum
            contactEmails={Array []}
            contactOnlyMode={undefined}
            platform=""
            role="dialog"
            title="Add Account"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "block": false,
            "className": undefined,
            "data-test": "add-account-modal-button",
            "disabled": false,
            "element": "button",
            "icon": undefined,
            "invalid": false,
            "label": "",
            "logic": false,
            "modaltrigger": true,
            "quiet": false,
            "selected": false,
            "variant": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "contactEmails": Array [],
            "contactOnlyMode": undefined,
            "platform": "",
            "role": "dialog",
            "title": "Add Account",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
